### ISSUE TITLE
Polymorphism via implements referencing external class not working

### ISSUE THREAD

**Issue by kanapin** [google/built_value.dart#1020]
This doesn't seem to compile:

```dart
// legs.dart 
// external class referenced from animal.dart, but not from cat.dart
class Legs {}

// animal.dart
import 'legs.dart';

abstract class Animal {
  Legs get legs;
}

// cat.dart
import 'animal.dart';

part 'cat.g.dart';

abstract class Cat implements Built<Cat, CatBuilder>, Animal {
  bool get tail;
}
```

Since `cat.dart` doesn't reference `legs.dart` class, neither does the `cat.g.dart`. So when it would try to compile `cat.g.dart`, it will fail with something like 

```
ERROR: cat.g.dart
final Legs legs;        Undefined class. #undefined_class
```

The workaround seems to re-export 'legs.dart' from 'animal.dart', or force import 'legs.dart' in 'animal.dart'.

**Comment by davidmorgan** [comment]
Yes, sorry about that. `part` files can't declare imports, so there's not much `built_value` can do here. Adding an export or an import is the right workaround.

Possibly we could warn you about that, but it would be nontrivial--we don't currently do any calculations around imports, we'd have to do quite a lot of work to figure out if an extra import is needed.

**Comment by davidmorgan** [comment]
(marking as a feature request; feature would be to do the computation around inputs and help the user add as needed)
